File: 백준/Silver/10828. 스택/스택.py
import sys
input = sys.stdin.readline

class stack:
    def __init__(self):
        self.list = []

    def push(self, x):
        self.list.append(x)
    
    def pop(self):
        len = self.size()
        if len == 0:
            return -1
        else:
            re = self.list[len-1]
            del self.list[len-1]
            return re

    def size(self):
        return len(self.list)

    def empty(self):
        len = self.size()
        if len == 0:
            return 1
        else:
            return 0

    def top(self):
        len = self.size()
        if len == 0:
            return -1
        else:
            return self.list[len-1]
        
n = int(input())
stack = stack()
for i in range(n):
    x = input().split()
    if x[0] == "push":
        stack.push(int(x[1]))
    elif x[0] == "pop":
        print(stack.pop())
    elif x[0] == "size":
        print(stack.size())
    elif x[0] == "empty":
        print(stack.empty())
    elif x[0] == "top":
        print(stack.top())

File: 백준/Silver/10828. 스택/스택.cc
#include <bits/stdc++.h>

using namespace std;

int main(int argc, char** argv)
{
    ios::sync_with_stdio(0);
    cin.tie(0);

    int n;
    cin >> n;
    string input;
    stack<int> S;
    for (int i = 0; i < n; i++) {
        cin >> input;
        if (input == "push") {
            int push;
            cin >> push;
            S.push(push);
        }
        else if (input == "pop") {
            if (!S.empty()) {
                cout << S.top(); 
                S.pop();
            }
            else cout << -1;
            cout << "\n";
        }
        else if (input == "size") {
            cout << S.size();
            cout << "\n";

        }
        else if (input == "empty") {
            if (S.empty()) cout << 1;
            else cout << 0;
            cout << "\n";
        }
        else if (input == "top") {
            if (!S.empty()) cout << S.top();
            else cout << -1;
            cout << "\n";
        }
    }

    return 0;
}